dumper/app/src/test/java/com/google/edwmigration/dumper/application/dumper/MetadataDumperTest.java

```diff
@@ -138,4 +140,62 @@ public void testDoesNotOverrideFileWithDirectory() throws IOException {
     // Assert
     assertTrue(exception.getMessage().startsWith("A file already exists at test"));
   }
+
+  @Test
+  public void testFailsOnUnrecognizedFlag() {
+    OptionException exception =
+        Assert.assertThrows(
+            "No exception thrown from " + dumper.getClass().getSimpleName(),
```

Comment:
Optional: This is not very important, but I think that adding this message doesn't help much. I would suggest removing it (also from the existing test in this class).

The difference between the test failing with and without the message is:

with:
```
No exception thrown from Main: expected java.lang.IllegalStateException to be thrown, but nothing was thrown
```

without:
```
expected java.lang.IllegalStateException to be thrown, but nothing was thrown
```